### PR TITLE
build(nym-connect-desktop): speedy mode

### LINKED ISSUE #3604: Hook up the speedy mixnet setup to NC UI and release it as NymConnect S (for speedy) on Github

### PR DESCRIPTION
**Pull request by doums**
### NC speedy mode

fix #3604

image

### CODE REVIEW

File: nym-connect/desktop/scripts/pre_meduim_build.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#! /bin/bash
```
Comment (octol) [inline nym-connect/desktop/scripts/pre_meduim_build.sh]: minor: filename spelling

File: nym-connect/desktop/scripts/pre_meduim_build.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+if ! [[ "$RELEASE_TAG" =~ ^nym-connect-s-v.*? ]]; then
+  echo -e " ✗ Invalid release tag $RELEASE_TAG"
+  exit 1
+fi
+
+version="${RELEASE_TAG#nym-connect-s-v}"
+
+sed -i "s/^name = \".*\"/name = \"nym-connect-s\"/" src-tauri/Cargo.toml
+sed -i "s/^version = \".*\"/version = \"$version\"/" src-tauri/Cargo.toml
+sed -i 's/"productName": "nym-connect"/"productName": "NymConnect S"/' src-tauri/tauri.conf.json
```
Comment (octol) [inline nym-connect/desktop/scripts/pre_meduim_build.sh]: will this cause issues? (the different capitalization)

**Review by octol — APPROVED** [review]
Rust parts seem fine to me!

**Comment by fmtabbara** [comment]
If we're going to be dealing with more than two modes it might be better to replace the `isMediumMode` boolean with something like "speedMode" which returns and string union e.g. `"slow" | "medium" | "fast".

But if we're only  concerned with two modes for now then `boolean`is fine.

**Review by fmtabbara — APPROVED** [review]
One general comment on the Conversation page about state.

File: nym-connect/desktop/scripts/pre_meduim_build.sh (hunk 2 above)
Comment (doums) [reply nym-connect/desktop/scripts/pre_meduim_build.sh]: I don't think, in the [doc](https://tauri.app/v1/api/config/#packageconfig) there are no notes regarding name convention

**Review by farbanas — APPROVED** [review]

**Pull request merged**